### PR TITLE
Contact Info Block: Split phone numbers into prefix and phone number

### PR DESCRIPTION
**Pull request by enejb**
This allows us to be not so strict with what a phone number is and still offer a way to prefix. things.

#### Changes proposed in this Pull Request
Previously we did text matching for things that looked liked phone numbers and auto converted them into phone numbers. This approach worked well moist times but it didn't in some cases. 

This changes take the phone number input and splits it into 2 parts. 
The first part is a prefix and the other part is a phone number. 

We then proceed and make the phone umber link-able. While leaving the prefix as it is.

#### Testing instructions
1. Add the contact info block. 
2. Try out different phone numbers. Does it work as expected?

### CODE REVIEW

File: client/gutenberg/extensions/contact-info/phone/save.js [hunk 1]
```diff
@@ -1,29 +1,43 @@
...
-					{ number }
-				</a>
-			);
+	// Assume that everthing after the first number should be a part of the phone number.
+	// care about the first prefix character.
+	let phoneNumber = indexOfFirstNumber ? inputText.substring( indexOfFirstNumber - 1 ) : inputText;
+	let prefix = indexOfFirstNumber ? inputText.substring( 0, indexOfFirstNumber ) : '';
+
+	let justNumber = phoneNumber.replace( /\D/g, '' );
+	// Phone numbers starting with + shoud be part of the number.
```
Comment (lezama) [inline client/gutenberg/extensions/contact-info/phone/save.js]:
```suggestion
	// Phone numbers starting with + should be part of the number.
```

File: client/gutenberg/extensions/contact-info/phone/save.js [hunk 2]
```diff
@@ -1,29 +1,43 @@
...
+		return inputText;
+	}
+	const indexOfFirstNumber = inputText.indexOf( arrayOfNumbers[ 0 ] );
 
-			return (
-				<a href={ `tel:${ justNumber }` } key={ i }>
-					{ number }
-				</a>
-			);
+	// Assume that everthing after the first number should be a part of the phone number.
```
Comment (lezama) [inline client/gutenberg/extensions/contact-info/phone/save.js]:
```suggestion
	// Assume that everything after the first number should be part of the phone number.
```

File: client/gutenberg/extensions/contact-info/phone/save.js [hunk 3]
```diff
@@ -1,29 +1,43 @@
...
+	// care about the first prefix character.
+	let phoneNumber = indexOfFirstNumber ? inputText.substring( indexOfFirstNumber - 1 ) : inputText;
+	let prefix = indexOfFirstNumber ? inputText.substring( 0, indexOfFirstNumber ) : '';
+
+	let justNumber = phoneNumber.replace( /\D/g, '' );
+	// Phone numbers starting with + shoud be part of the number.
+	if ( /[0-9/+/(]/.test( phoneNumber.substring( 0, 1 ) ) ) {
+		// Remove the special charater from the prefix so they don't appear twice.
```
Comment (lezama) [inline client/gutenberg/extensions/contact-info/phone/save.js]:
```suggestion
		// Remove the special character from the prefix so they don't appear twice.
```

File: client/gutenberg/extensions/contact-info/phone/save.js (hunk 1 above)
Comment (enejb) [reply client/gutenberg/extensions/contact-info/phone/save.js]: fixed

File: client/gutenberg/extensions/contact-info/phone/save.js (hunk 2 above)
Comment (enejb) [reply client/gutenberg/extensions/contact-info/phone/save.js]: fixed

File: client/gutenberg/extensions/contact-info/phone/save.js (hunk 3 above)
Comment (enejb) [reply client/gutenberg/extensions/contact-info/phone/save.js]: fixed

**Comment by jeherve** [comment]
Cherry-picked to the release branch in 7ca9928b93c3f2037cf377044818dbbd409fd845

**Pull request merged**